server/models.py

```diff
@@ -0,0 +1,65 @@
+from database import db
+from sqlalchemy.orm import validates
+from datetime import timedelta, datetime
+
+
+class User(db.Model):
+    # dummy user model for now that will allow us to create contests
+    __tablename__ = 'user'
+
+    id = db.Column(db.Integer, primary_key=True)
+
+    def __init__(self, id):
+        self.id = id
+
+    def __repr__(self):
+        return f'User number {self.id}'
+
+
+class Contest(db.Model):
+    __tablename__ = 'contest'
+
+    id = db.Column(db.Integer, primary_key=True)
+
+    # should we force title to be unique or no?
+    title = db.Column(db.String(100), unique=True, nullable=False)
+    description = db.Column(db.Text)
+
+    # need to decide which fields should allow null values
+    prize = db.Column(db.Float, nullable=False)
+    deadline = db.Column(db.DateTime, nullable=False)
+    date_created = db.Column(db.DateTime, nullable=False)
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
+
+    def __repr__(self):
+        return f'Contest {self.id}, Title: {self.title}'
+
+    # example validation making sure deadline for contest is
+    #   within the next year, need to decide on what to validate
+    #   for both models
+
+    @validates('deadline')
+    def validate_deadline(self, key, deadline):
+        # take other action besides assert
+        assert deadline < datetime.now() + timedelta(days=365)
+        return deadline
```

Comment:
looks great overall! one suggestion i have is to allow your model to determine the JSON structure for you so that you don't have to do that in your controller. 

for example you can add and override the `to_dict` method here in this `Contest` model. then, in the controller, you can just call `Contest.to_dict()`

that may reduce many number lines of code in the future :)